self-tests.yml

```diff
@@ -389,6 +390,29 @@ tasks:
           target: "coverage-html"
           make_args: "-k"
           tz: "America/New_York"
+
+  - name: staging-dist 
```

Comment:
Could you do this by calling "run-build" in a way similar to the way the dist task is defined and parameterize the values that are different? I suspect you could reduce some amount of code duplication here.

Reply:
I considered this, but as far as I can tell the only way to get a variable into the &run-build template is to use the $task_name or other default expansions. I would hardwire the make target to 'dist' and use $task_name to determine between staging and not, but I think this would require the task name to be empty, or parsing of the task name somehow. I also don't think I can convert run-build into a function, as it is a collection of functions. I think the only way to do this is create separate make targets: dist and dist-staging, which I'm not sure how to do without creating other problems (but am happy to figure out if you think it's a good idea, I don't have any experience with makefiles)

Reply:
Ah, that makes sense. I am curious about creating a separate make target. Which problems might that cause?

Reply:
Not really sure. I am wary of having an environment variable and a target that are both supposed to do the same thing, but I don't actually know whether or not that's a totally normal thing to do with makefiles.

Reply:
Could you remove the environment variable entirely, and instead have the two targets set a make variable, and then call some single function that does the rest of the work?

Reply:
I think I have something that works, and I've tested it in staging. The speedup is noticeable!